src/aead.rs

```diff
@@ -664,6 +664,76 @@ enum Direction {
     Sealing,
 }
 
+trait BitXor {
+    fn bitxor_assign(&mut self, input: [u8; BLOCK_LEN]);
+}
+
+impl BitXor for [u8; BLOCK_LEN] {
```

Comment:
I understand you can't implement `core::ops::BitXorAssign` for [u8; BLOCK_LEN]` due to the coherence rules. But, maybe we just need a function: `bitxor_assign(r: &mut [u8; BLOCK_LEN], a: [u8; BLOCK_LEN]) { ... }` instead of a whole trait?

Reply:
> maybe we just need a function:bitxor_assign(r: &mut [u8; BLOCK_LEN], a: [u8; BLOCK_LEN]) { ... }` instead of a whole trait?

Probably just the function would be fine. My idea was to separate into traits for modularity/reuse (maybe something else would need it). Another option would be once `Block` struct is removed, create a new trait with the good bits pulled from `Block` that we still want to keep around.

Wasn't sure what grouping made sense, though, so separate traits it was.